DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -70,38 +67,38 @@ public async Task<IActionResult> Index(LoginViewModel model)
             var (verifiedAdminUser, verifiedDelegateUsers) =
                 loginService.VerifyUsers(model.Password!, adminUser, delegateUsers);
 
-            if (adminUser != null && adminUser.IsLocked)
+            var adminAccountFailedVerification = adminUser != null && verifiedAdminUser == null;
+            var adminAccountIsLocked = adminUser != null && adminUser.IsLocked ||
+                                       adminAccountFailedVerification && adminUser!.FailedLoginCount == 4;
+            var shouldLogIntoDelegateAccount = verifiedDelegateUsers.Any();

```

Comment:
I know I came up with these names, but looking at where they're used I think they could be improved.

I think this should be changed to `delegateAccountVerificationSuccessful`, and I'd change `adminAccountFailedVerification` to `adminAccountVerificationAttemptedAndFailed` to make it clear that it's not just "There's no admin account to log into". Does that sound reasonable?

Reply:
yeah those sound reasonable to me.